nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/QueryMarkLogic.java

```diff
@@ -198,26 +199,32 @@ protected Collection<ValidationResult> customValidate(final ValidationContext va
      * @throws ProcessException
      */
     @Override
+    // TODO: Ask Rob how to pass the flow file to success areas.
+    // TODO: Ask Rob how to get a toString representation of the query definition that was run by the processor and save it to the attribute.
     public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFactory) throws ProcessException {
         final ProcessSession session = sessionFactory.createSession();
         super.populatePropertiesByPrefix(context);
-        try {
-            final FlowFile incomingFlowFile = context.hasIncomingConnection() ? session.get() : null;
 
-            Tuple<DataMovementManager, QueryBatcher> tuple = newQueryBatcher(context, incomingFlowFile);
-            configureQueryBatcher(context, session, incomingFlowFile, tuple.getValue());
+        FlowFile incomingFlowFile = session.get();
+         if (incomingFlowFile == null) {
+             incomingFlowFile = session.create();
+         }
+
+        try {
+            QueryBatcherContext  queryBatcherContext = newQueryBatcher(context, incomingFlowFile);
+            session.putAttribute(incomingFlowFile, "marklogic-query", queryBatcherContext.getDefinition().toString());
```

Comment:
Sorry for the delay on this - I did some testing of this, and clearly `toString()` doesn't return anything helpful. And there's unfortunately not a method in the `QueryDefinition` interface that can return something useful. 

But, in the code in `QueryBatcherBuilder` that constructs a `QueryDefinition`, we know how to capture each type of query as a string such that this attribute will contain something useful:

1. If stateIndexRangeQuery is not null, then the string representation is `String rawCombinedQuery`
2. If query is RawCombinedQueryDefinition, can call `getHandle().toString()` on that to get a string representation
3. If query is RawStructuredQueryDefinition, can call `serialize()` on that to get a string representation
4. If query is StructuredQueryDefinition, can call `serialized()` on that to get a string representation
5. If query is StringQueryDefinition, can call `getCriteria()` on that to get a string representation

Thus, we should add a `String queryRepresentation` to `QueryBatcherContext` and have `newQueryBatcher` populate it based on the logic above. This line of code can then just call `queryBatcherContext.getQueryRepresentation()`.